### PR TITLE
Adding get_credits_balance and get_credits_history

### PR DESCRIPTION
**Pull request by andher1802**
Adding the get_credits_balance and get_credits_history on the tools Class.

Items:
* [ x] Ran test & live-tests
* [ x] Implemented (new) unit tests
* [ ] Removed credentials
* [ ] Removed argument pointing to staging
* [ ] Updated [documentation](sdk.up42.com)

For release:
* [ ] Bumped version
* [ ] Added changelog
* [ ] Updated announcement banner

### CODE REVIEW

**Comment by andher1802** [comment]
@chrieke this is a preliminary PR for keep you updated about progress and possible feedback for further reference. I need to improve the testing and also add local tests to the get_credits_history function.

File: tests/test_tools.py [hunk 1]
```diff
@@ -104,6 +105,55 @@ def test_get_block_details_live(tools_live):
...
+        start_date="2022-03-01", end_date="2022-03-02"
+    )
+    assert isinstance(balance_history, dict)
+    assert "content" in balance_history
+    assert len(balance_history["content"]) == 32
+    assert isinstance(balance_history["content"], list)
+
+
+@pytest.mark.live
+def test_get_credits_history_no_bounds_live(tools_live):
```
Comment (chrieke) [inline tests/test_tools.py:142]: What this is really checking is that the function code can handle a not defined start (or enddate), not so much the content of the response. Therefore could also just be a mocktest. Ideally could be using @pytest.parameterize .

File: tests/test_tools.py [hunk 2]
```diff
@@ -104,6 +105,55 @@ def test_get_block_details_live(tools_live):
...
+
+
+@pytest.mark.live
+def test_get_credits_history_live(tools_live):
+    balance_history = tools_live.get_credits_history(
+        start_date="2022-03-01", end_date="2022-03-02"
+    )
+    assert isinstance(balance_history, dict)
+    assert "content" in balance_history
+    assert len(balance_history["content"]) == 32
```
Comment (chrieke) [inline tests/test_tools.py:137]: Could be that in the future this fails if test runs add additional items to the credit history of the sdk test account, maybe do `>= 32` Allthough weird that it is only 32 overall hm.

File: up42/tools.py [hunk 3]
```diff
@@ -224,6 +223,88 @@ def get_block_details(self, block_id: str, as_dataframe: bool = False) -> dict:
...
+        account.
+        Args:
+            start_date: The start date for the credit consumption search e.g.
+            2021-12-01. Default start_date None uses 2000-01-01.
+            end_date: The end date for the credit consumption search e.g.
+            2021-12-31. Default end_date None uses current date.
+
+        Returns:
+            A dict with the information of the credit consumption records for
+            all the users linked by the account_id. The result match the endpoint
```
Comment (chrieke) [inline up42/tools.py:259]: "The result match the endpoint response with all records on the content key."
Not sure what this means, possible to clarify or remove?

File: up42/tools.py [hunk 4]
```diff
@@ -224,6 +223,88 @@ def get_block_details(self, block_id: str, as_dataframe: bool = False) -> dict:
...
+            end_date = tomorrow_datetime.strftime("%Y-%m-%d")
+
+        [start_formatted_date, end_formatted_date] = format_time_period(
+            start_date=start_date, end_date=end_date
+        ).split("/")
+        search_parameters = dict(
+            {
+                "from": start_formatted_date,
+                "to": end_formatted_date,
+                "size": 2000,
```
Comment (chrieke) [inline up42/tools.py:286]:
```suggestion
                "size": 2000,  # 2000 is the maximum page size for this call
```

File: up42/tools.py [hunk 5]
```diff
@@ -224,6 +223,88 @@ def get_block_details(self, block_id: str, as_dataframe: bool = False) -> dict:
...
+        )
+        endpoint_url = f"{self.auth._endpoint()}/accounts/me/credits/history"
+        response_json: dict = self.auth._request(
+            request_type="GET", url=endpoint_url, querystring=search_parameters
+        )
+        isLastPage = response_json["data"]["last"]
+        credit_history = response_json["data"]["content"]
+        result = dict(response_json["data"])
+        del result["content"]
+        while not isLastPage:
```
Comment (chrieke) [inline up42/tools.py:298]: We should probably add a test for the pagination, a bit similar to [here](https://github.com/up42/up42-py/blob/88a34b05d4606811943dd41e10078f29da551d3c/tests/test_catalog.py#L290), also see the pagination_mock fixture.

**Review by chrieke — COMMENTED** [review]
Nice! A few small comments. Probably good to add a mock test for the pagination, as due to the low order volume on the sdk test account this can not be tested in live scenario.

File: tests/test_tools.py (hunk 2 above)
Comment (andher1802) [reply tests/test_tools.py:136]: I fixed the inputs for being just one day. Therefore, I think we should not expect changes as there could not be more transactions on this time range. But, maybe if there is some adjust on the database and they include new credits consumed on this date, then we could have an error.

Comment (chrieke) [reply tests/test_tools.py:136]: ah ja, forgot about the fixed dates.

File: tests/fixtures.py [hunk 6]
```diff
@@ -234,6 +234,11 @@
     },
     "error": None,
 }
+
+JSON_BALANCE = {"data": {"balance": 10693}}
+
+JSON_CREDITS_HISTORY = JSON_ORDERS
```
[the remark this answers is not in the document]
Comment (andher1802) [reply tests/fixtures.py]: Done!

File: tests/test_tools.py (hunk 1 above)
Comment (andher1802) [reply tests/test_tools.py]: Done!

File: tests/test_tools.py (hunk 2 above)
Comment (andher1802) [reply tests/test_tools.py:150]: Checked and Done!

File: up42/tools.py (hunk 5 above)
Comment (andher1802) [reply up42/tools.py:298]: Done! Not the same approach than the catalog pagination test, but similar idea.

**Pull request merged**